packages/node-core/src/indexer/poi/poi.service.ts

```diff
@@ -29,9 +30,12 @@ export class PoiService implements OnApplicationShutdown {
   private _poiRepo?: CachePoiModel;
   private _latestSyncedPoi?: ProofOfIndex;
   private isSyncing = false;
-  private _projectId?: string;
 
-  constructor(private storeCache: StoreCacheService, private eventEmitter: EventEmitter2) {}
+  constructor(
+    private storeCache: StoreCacheService,
+    private eventEmitter: EventEmitter2,
+    private project: ISubqueryProject
```

Comment:
You can use `@Inject('ISubqueryProject')` here instead of having to use a factory